plugins/baseball-scoreboard/logo_manager.py

```
@@ -183,6 +185,82 @@ def load_milb_logo(self, team_abbr: str, logo_dir: Path) -> Optional[Image.Image
+        if url and str(url).lower().startswith(("http://", "https://")):
+            try:
+                resp = requests.get(
+                    url, timeout=5, headers={"User-Agent": "LEDMatrix Baseball Plugin/1.0"}
+                )
+                resp.raise_for_status()
+                full = Image.open(BytesIO(resp.content)).convert("RGBA")
+                if disk_path is not None:
+                    try:
+                        disk_path.parent.mkdir(parents=True, exist_ok=True)
+                        full.save(disk_path, "PNG")
+                    except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)